zivid_camera/src/settings_generator.cpp

```diff
@@ -240,6 +290,16 @@ class DynamicReconfigureCfgGenerator
     const auto type_name = rosTypeName<decltype(convertValueToRosValue(default_value))>();
     const auto default_value_str = valueTypeToRosTypeString(default_value);
 
+    if constexpr (IsEnumSetting<SettingsNode>::value)
+    {
+      const auto valid_values = node.validValues();
+      std::vector<std::string> enum_constants;
+      std::transform(valid_values.cbegin(), valid_values.cend(), std::back_inserter(enum_constants),
+                     [&](const auto value) { return generateEnumConstant<SettingsNode>(value); });
+      ss_ << rosGeneratedEnumVariableName(setting_name) << " = gen.enum([\n    "
```

Comment:
I guess this `rosGeneratedEnumVariableName(setting_name)` has to be equal to the one you pass to `edit_emthod=` further down. Would be nice to save the name in a variable and use that both places, for DRY.
But I guess that can be difficult with the if constexpr's, i.e. we don't know that the setting is a enum?

Reply:
Yeah, thats why I made it a function that is called from two places.

Reply:
Yes, looks good to me